src/main/java/com/transferwise/openbanking/client/json/JacksonJsonConverter.java

```diff
@@ -43,7 +43,9 @@ public String writeValueAsString(Object value) {
         try {
             return objectMapper.readValue(content, valueType);
         } catch (JsonProcessingException e) {
-            throw new ClientException("Unable to read JSON string to object", e);
+            throw new ClientException(
+                "Unable to read " + valueType.getSimpleName() + " object from JSON string '" + content + "'",
```

Comment:
With this there is a risk of logging or otherwise exposing potentially sensitive data, say for example this class gets used to de-serialize a string that contains a client secret field. 

I considered two mitigations for this: 

- throw a more specific exception here, say a `JsonParsingException`, so that a caller of the library can catch this specific exception and say change their logging call in this situation
- include the JSON string as a property within the exception, and not in the message, so that a caller can explicitly access the value and log it if they want (or not)

I didn't end up doing either mainly because: 

- the Jackson library will include the full string in the exception message anyway, for certain kinds of invalid JSON (e.g. a value of just `ABC`)
- on invalid JSON, we will most likely always want to log or view in some way the full value
- there is limited sensitive information in the payment API responses

Happy to hear other thoughts on this though.

Reply:
For this case I don't have other options. One concert is that the error will be also present in Rollbar and not only logged in our Kibana infrastructure

Reply:
I ended up having a bit more of a think and went with the first mitigation mentioned above, adding new specific exceptions. For us it won't complicate things much and it's probably better to by default not include the JSON in the message.

Reply:
Having dedicated exception class make sense. Looks good to me.